Assets/LeapMotion/Modules/HierarchyRecording/Scripts/LeapRecording/LeapRecording.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.Recording {
+
+  public abstract class LeapRecording : ScriptableObject {
+
+    /// <summary>
+    /// Converts nanoseconds to seconds. Leap frames store timestamps in nanoseconds.
+    /// </summary>
+    public const double NS_TO_S = 1e-6;
+
+    /// <summary>
+    /// Converts seconds to nanoseconds. Leap frames store timestamps in nanoseconds.
+    /// </summary>
+    public const double S_TO_NS = 1e6;
+
+    /// <summary>
+    /// Returns the length of the recording in seconds.
+    /// </summary>
+    public abstract float length { get; }
+
+    /// <summary>
+    /// Loads this recording with data from the provided list of frames. These frames
+    /// reflect raw recorded Leap data. The actual frame storage scheme utilized is up to
+    /// the implementation.
+    /// 
+    /// If this recording already has frame data, a call to this method should overwrite
+    /// that data.
+    /// </summary>
+    public abstract void LoadFrames(List<Frame> frames);
```

Comment:
We should be specific in the description of this method whether or not the argument list can be expected to be destroyed or kept around by the recording.  Currently the raw recording just holds onto it, but that might have unexpected results if the user goes off and does something else with their list later.

Reply:
Good call. I feel like a copy is more appropriate. I personally like to use pooled lists all the time and would prefer not to require that the input list is persistent. Easy enough change for the raw recording implementation. What do you think?

Reply:
I agree, I think a copy is the way to go.  And I don't think the performance overhead of a copy will every be anything to worry about, since the cost of saving a recording in the first place will be pretty large.

Reply:
pushed copying